lib/qesapdeployment.pm

```diff
@@ -2353,20 +2354,23 @@ sub qesap_terraform_clean_up_retry {
     Return 0: we manage the failure properly
     Return 1: something went wrong or we do not know what to do with the failure
 
-=over 1
+=over 2
 
 =item B<ERROR_LOG> - error log filename
 
+=item B<PROVIDER> - cloud provider name as from PUBLIC_CLOUD_PROVIDER setting
+
 =back
 =cut
 
 sub qesap_terrafom_ansible_deploy_retry {
     my (%args) = @_;
-    croak 'Missing mandatory error_log argument' unless $args{error_log};
+    foreach (qw(error_log provider)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+
+    my $detected_error = qesap_ansible_error_detection(error_log => $args{error_log});
     my @ret;
 
-    if (qesap_file_find_string(file => $args{error_log}, search_string => 'Missing sudo password')) {
-        record_info('DETECTED ANSIBLE ERROR', 'MISSING SUDO PASSWORD');
+    if ($detected_error eq 3) {
```

Comment:
It's already merged, so no need to do anything but for next time: `eq` for strings and `==` for numbers.

Generally there are more issues when `==` is used in place of `eq` (not the case here), but at the very least what is happening here internally is that 3 (an integer) and `$detected_error` (also an integer) are both being converted to string before the comparison.